Render fenced block how it wants to be rendered

GuySartorelli opened the pull request:
The `FencedCode` class stores information about what delimiter character should be used, how many should be used, and what the indent offset is for the block.


This is important information to retain and re-use, since a fenced block indicating markdown might look like:
`````
````
This is how a fenced block looks:
```php
<?
// some code here
```
````
`````

In this example, the four backticks are used to denote our actual fenced code block, and the three backticks are part of the literal code inside the block. Without retaining the information about how many delimiter characters to use, you might end up closing the block early.


The offset is useful because you might have a fenced code block inside a list like so:
````
- some random thing
  ```
  This fenced block is part of the list item, so its indent is important to avoid cancelling the list and starting a new one
  ```
- another thing
````

Finally, reusing the original delimiter character is useful. Tildes are a valid character to use. If someone uses this library to modify part of their markdown AST and then write back the file, it would be nice to retain their original code block delimiters so they don't have to go swapping them back to what they used to be.

## NOTE
There's no issues in this repo so I can't open an issue about it - but there are other nodes as well (such as list items) which tell you what character should be used to render it. This library would be so much more useful if it retained the original options. If your intention with this renderer is that it normalises things, then having a configuration option to use the originals instead of normalisation would be useful.

stefanzweifel commented:
Thanks for this PR @GuySartorelli! ♥️

> This library would be so much more useful if it retained the original options.

Wholeheartedly agree on this. When I developed v1 of this package, this was the big downside of my approach. I just didn't know how I could extract the used characters or how I should calculate offsets.

> If your intention with this renderer is that it normalises things, then having a configuration option to use the originals instead of normalisation would be useful.

Normalising input/output was not my ittention. It's _just_ a side effect of how the code works. I'm open for suggestions on how to make this package better.

> There's no issues in this repo so I can't open an issue about it […]

I've just enabled Issues and Discussions on this repo. Feel free to open issues for the bugs you have found.

---

Will try to take a closer look at your PR and why the tests are failing in the next few days.

GuySartorelli commented:
Awesome, thanks for your feedback. I'll see if I can find all the other nodes which have properties explaining how they should be rendered and create an issue for them.

For this PR, I've fixed the issue with the test and updated the tests to include new scenarios for the new functionality. Let me know if you want any other changes.

stefanzweifel reviewed (approved):
Sorry for my late review/answer here.
LGTM.

Will create issues/todos for the other nodes shortly.
